internal/gitinterface/log_test.go

```diff
@@ -102,4 +99,218 @@ func TestGetCommitsBetweenRange(t *testing.T) {
+		}
+	})
+
+	t.Run("Test commit 4", func(t *testing.T) {
+		commits, err := GetCommitsBetweenRange(repo, commitIDs[3], plumbing.ZeroHash)
+		assert.Nil(t, err)
+		// commit 4 is the child of commit 2, so only it, commit 2, and commit 2's parent commit 1 should be returned
+		expectedCommits := []*object.Commit{allCommits[1], allCommits[0], allCommits[3]}
+		if !reflect.DeepEqual(expectedCommits, commits) {
+			t.Errorf("Expected and actual commits do not match. Expected: %v, Got: %v", expectedCommits, commits)
+		}
+	})
+
+	t.Run("Test commit 5, the merge commit", func(t *testing.T) {
+		// commit 5 is the merge commit of commit 2 and commit 3, so it should return commit 5, commit 2, commit 3, and commit 1 (the parent of commit 2 and commit 3)
+		commits, err := GetCommitsBetweenRange(repo, commitIDs[4], plumbing.ZeroHash)
+		assert.Nil(t, err)
+		expectedCommits := []*object.Commit{allCommits[4], allCommits[1], allCommits[0], allCommits[2]}
+		if !reflect.DeepEqual(expectedCommits, commits) {
+			t.Errorf("Expected and actual commits do not match. Expected: %v, Got: %v", expectedCommits, commits)
+		}
+	})
+
+	t.Run("Test commit 6", func(t *testing.T) {
+		// commit 6 is the child of commit 3, so it should return commit 6, commit 3, and commit 1 (the parent of commit 3)
+		commits, err := GetCommitsBetweenRange(repo, commitIDs[5], plumbing.ZeroHash)
+		assert.Nil(t, err)
+		expectedCommits := []*object.Commit{allCommits[0], allCommits[5], allCommits[2]}
+		if !reflect.DeepEqual(expectedCommits, commits) {
+			t.Errorf("Expected and actual commits do not match. Expected: %v, Got: %v", expectedCommits, commits)
+		}
+	})
+}
+
+func GetCommitsFromCommitIDs(commitIDs []plumbing.Hash, repo *git.Repository) ([]*object.Commit, error) {
+	allCommits := make([]*object.Commit, 0, len(commitIDs))
+	for _, commitID := range commitIDs {
+		commit, err := GetCommit(repo, commitID)
+		if err != nil {
+			return nil, err
+		}
+
+		allCommits = append(allCommits, commit)
+	}
+
+	return allCommits, nil
+}
+
+func createTestTrees(t *testing.T, repo *git.Repository, emptyBlobHash plumbing.Hash, num int) []plumbing.Hash {
+	t.Helper()
+	treeHashes := make([]plumbing.Hash, 0, num)
+	for i := 1; i <= num; i++ {
+		objects := make([]object.TreeEntry, 0, i)
+		for j := 0; j < i; j++ {
+			objects = append(objects, object.TreeEntry{Name: fmt.Sprintf("%d", j+1), Hash: emptyBlobHash})
+		}
+
+		treeHash, err := WriteTree(repo, objects)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		treeHashes = append(treeHashes, treeHash)
+	}
+	return treeHashes
+}
+
+func createChildrenCommits(t *testing.T, repo *git.Repository, treeHashes []plumbing.Hash, numChildren int) []plumbing.Hash {
+	t.Helper()
+
+	children := make([]plumbing.Hash, 0, numChildren)
+
+	for i := 1; i <= numChildren; i++ {
+		commit := CreateCommitObject(testGitConfig, treeHashes[i], []plumbing.Hash{}, fmt.Sprintf("Test commit %v", i+1), testClock)
```

Comment:
this also ends up with no parents, you want to pass in the parent commit ID(s) here.

Reply:
@adityasaky I have done this, is this the change you mean?